src/kw_include.sh

```diff
@@ -0,0 +1,25 @@
+# Used for file sourcing,
+# accepts as argument the path for the file to be sourced
+function include()
+{
+  local filename="$1"
+  if [[ ! -e "$filename" ]]; then
+    echo "File $filename could not be found, check your file path."
+    return 2
+  fi
+
+  if [[ ! -r "$filename" ]]; then
+    echo "File $filename could not be read, check your file permissions."
+    return 1
+  fi
+
+  local varname=$(basename "$filename" .sh)
+  varname=${varname^^}_IMPORTED # capitalize and append "_IMPORTED"
+
+  if [[ -v "${varname}" ]]; then
+    return 0
+  else
+    declare -g "${varname}"=1
+    . "$filename" --source-only # can return 0, 1 or 2
```

Comment:
From bash documentation:
> If filename is not found, or cannot be read, the return status is non-zero.

In this sense, you already checked the status in the previous if condition, so... no need for the extra comment.